### PR TITLE
React Query-2 : added query for auth

### PR DESCRIPTION
**Pull request by sahsisunny**
## Issue Ticket Number

-   #56

## Description

Added query and modify hooks for auth

**Breaking Changes**

-   [ ] Yes
-   [x] No

**Development Tested?**

-   [x] Yes
-   [ ] No

**Tested in Staging?**

-   [ ] Yes
-   [x] No

**Under Feature Flag**

-   [ ] Yes
-   [x] No

_Specify if the changes are currently under a feature flag._

**Database Changes**

-   [ ] Yes
-   [x] No

### Screenshots

https://github.com/Real-Dev-Squad/tiny-site-frontend/assets/70854507/92dbb2d0-0c17-4894-9977-0a3faef86186

## Test Coverage
| Component                   | Statements | Branches | Functions | Lines | 
|-----------------------------|------------|----------|-----------|-------|
| `src/components/Navbar`     |   94.11    |   100    |   71.42   | 93.54 |
| `NavbarMenuItems.tsx`       |   100      |   100    |   100     | 100   |
| `UserProfileButton.tsx`     |   100      |   100    |   100     | 100   |
| `index.tsx`                 |   90.47    |   100    |   50      | 90    |
| `src/hooks/useAuthenticated.ts` | 100    |   100    |   100     | 100   |

## Additional Notes

N/A

### CODE REVIEW

File: __tests__/components/Navbar.test.tsx [hunk 1]
```diff
@@ -1,19 +1,57 @@
...
 
 import Navbar from '@/components/Navbar/';
 
+import user from '../../__mocks__/db/user';
+import handlers from '../../__mocks__/handler';
+
+const server = setupServer(...handlers);
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
```
Comment (vinit717) [inline __tests__/components/Navbar.test.tsx:16]: No need to to this you can remove it

File: src/services/api.tsx
Comment (vinit717) [inline src/services/api.tsx:1]: Can we not test this file?

**Comment by vinit717** [comment]
Please add tests coverage also

File: src/services/api.tsx
Comment (sahsisunny) [reply src/services/api.tsx:1]: https://github.com/Real-Dev-Squad/tiny-site-frontend/blob/d54d949de74e04fc980dd6e647c9f6e28c828620/__tests__/hooks/api.test.tsx

File: __tests__/hooks/useAuthenticated.test.tsx [hunk 2]
```diff
@@ -1,46 +1,40 @@
...
 
-import { userData } from '../../fixtures/users';
+import user from '../../__mocks__/db/user';
+import handlers from '../../__mocks__/handler';
 
-jest.mock('axios');
+const server = setupServer(...handlers);
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply __tests__/hooks/useAuthenticated.test.tsx]: Removed

**Comment by ajoykumardas12** [comment]
Please add test stats.
Also, change under feature flag is said yes but don't see feature flag.
Tested in staging should be no I think?

**Review by ajoykumardas12 — APPROVED** [review]
LGTM

**Review by vinit717 — APPROVED** [review]

**Pull request merged**